tests/suite/test_virtual_server_mixed_grpc.py

```diff
@@ -0,0 +1,126 @@
+                                backend_setup, virtual_server_setup):
+        print("\nStep 1: assert config")
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            virtual_server_setup.namespace,
+                                            virtual_server_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+
+        assert "proxy_connect_timeout 60s;" in config
+        assert "proxy_read_timeout 60s;" in config
+        assert "proxy_send_timeout 60s;" in config
+
+        assert "proxy_set_header Upgrade $http_upgrade;" in config
+        assert "proxy_http_version 1.1;" in config
+
+        assert "proxy_next_upstream error timeout;" in config
+        assert "proxy_next_upstream_timeout 0s;" in config
+        assert "proxy_next_upstream_tries 0;" in config
+
+        assert "grpc_connect_timeout 60s;" in config
+        assert "grpc_read_timeout 60s;" in config
+        assert "grpc_send_timeout 60s;" in config
+
+        assert "grpc_set_header X-Real-IP $remote_addr;" in config
+        assert "grpc_set_header X-Forwarded-For $proxy_add_x_forwarded_for;" in config
+        assert "grpc_set_header X-Forwarded-Host $host;" in config
+        assert "grpc_set_header X-Forwarded-Port $server_port;" in config
+        assert "grpc_set_header X-Forwarded-Proto $scheme;" in config
+
+        assert 'grpc_set_header Host "$host";' in config
+
+        assert "grpc_next_upstream error timeout;" in config
+        assert "grpc_next_upstream_timeout 0s;" in config
+        assert "grpc_next_upstream_tries 0;" in config
+
+        print("\nStep 2: check connection to http backend")
+        resp = requests.get(virtual_server_setup.backend_2_url, headers={"host": virtual_server_setup.vs_host})
+        print("Response from http backend: {}".format(resp))
+        assert resp.status_code == 200
+
+        print("\nStep 2: Check connection to app")
+        cert = get_certificate(virtual_server_setup.public_endpoint.public_ip,
+                               virtual_server_setup.vs_host,
+                               virtual_server_setup.public_endpoint.port_ssl)
+        target = f'{virtual_server_setup.public_endpoint.public_ip}:{virtual_server_setup.public_endpoint.port_ssl}'
+        credentials = grpc.ssl_channel_credentials(root_certificates=cert.encode())
+        options = (('grpc.ssl_target_name_override', virtual_server_setup.vs_host),)
+
+        with grpc.secure_channel(target, credentials, options) as channel:
+            stub = GreeterStub(channel)
+            response = ""
+            try:
+                response = stub.SayHello(HelloRequest(name=virtual_server_setup.public_endpoint.public_ip))
```

Comment:
is it possible to assert something based on the data in response?

Reply:
Done